src/fiber.cr

```diff
@@ -46,9 +45,21 @@ class Fiber
 
     fiber_main = ->(f : Fiber) { f.run }
 
-    # point to first addressable pointer on the stack (@stack_bottom points past
-    # the stack because the stack grows down):
-    stack_ptr = @stack_bottom - sizeof(Void*)
+    # FIXME: This line shouldn't be necessary
+    stack_ptr = nil
+    {% if flag?(:win32) %}
+      # align stack bottom to 16 bytes
+      @stack_bottom = Pointer(Void).new(@stack_bottom.address & ~0x0f_u64)
+
+      # It's the caller's responsibility to allocate 32 bytes of "shadow space" on the stack right
+      # before calling the function (regardless of the actual number of parameters used)
+
+      stack_ptr = @stack_bottom - sizeof(Void*) * 4
```

Comment:
I have been thinking a bit about the shadow space, since at first I didn't think that we needed it at all. I have concluded that we do need it but I think the way we do it here gives the wrong end result. 

Right now we first reserve 4 bytes of shadow space, and then write `rsp` over one of them when writing `stack_ptr[0] = fiber_main.pointer`. I think we have to do like I show in this diagram since the next 16 byte aligned offset is 48 bytes from the bottom of the stack.
<img width="431" alt="Skjermbilde 2019-07-26 kl  23 25 14" src="https://user-images.githubusercontent.com/8337848/61986188-bdfc4200-b00d-11e9-957c-0f791c998026.png">
(it's just exemplified with "addresses" on a 64 byte stack).

The fact that we don't have a return address should not matter, the important part is that `rsp` ends up in the shadow space which might be used when debugging for example. `stack_ptr = @stack_bottom - sizeof(Void*) * 6` should avoid that.

Take a look at how [Boost Context does it](https://github.com/boostorg/context/blob/d4608a4e8ba2e8a0bdd45613243e6c53a23d23dc/src/asm/ontop_x86_64_ms_pe_gas.asm#L190). I linked to the place int the asm where they set up`rsp` and compare that offset with the position in the chart in the top of the file.
 
@ysbaddaden had some questions about the shadow space earlier and i have some references here besides the Boost Context one:
https://docs.microsoft.com/en-us/cpp/build/stack-usage?view=vs-2019#stack-allocation
https://github.com/simon-whitehead/assembly-fun/tree/master/windows-x64#shadow-space
https://wiki.lazarus.freepascal.org/Win64/AMD64_API#Shadow_space